### PR TITLE
add redis cache demo for cookie session sample

### PR DESCRIPTION
**Pull request by xyting**

### CODE REVIEW

File: samples/CookieSessionSample/RedisCacheTicketStore.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.Cookies;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Redis;
+
+namespace CookieSessionSample 
+{
+    public class RedisCacheTicketStore : ITicketStore
```
Comment (Tratcher) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: Why Redis and not just a DistributedCacheTicketStore?

Comment (xyting) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: good, thanks

File: samples/CookieSessionSample/RedisCacheTicketStore.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        private static string GetUniqueKey() 
+        {
+            int maxSize = 8;
+            int size = maxSize;
+            var data = new byte[1];
+            var crypto = new RNGCryptoServiceProvider();
+            crypto.GetNonZeroBytes(data);
+            size = maxSize;
```
Comment (dallancarr) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: double assignment here 😊

File: samples/CookieSessionSample/RedisCacheTicketStore.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        {
+            await _cache.RemoveAsync(key);
+        }
+
+        private static string GetUniqueKey() 
+        {
+            int maxSize = 8;
+            int size = maxSize;
+            var data = new byte[1];
+            var crypto = new RNGCryptoServiceProvider();
```
Comment (PinpointTownes) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: `RNGCryptoServiceProvider` will only work on Windows and Mono. Consider replacing it by `RandomNumberGenerator.Create()` to get a x-plat'-friendly RNG.

File: samples/CookieSessionSample/RedisCacheTicketStore.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+            return ticket;
+        }
+
+        public async Task RemoveAsync(string key) 
+        {
+            await _cache.RemoveAsync(key);
+        }
+
+        private static string GetUniqueKey()
```
Comment (PinpointTownes) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: This method seems overly complicated. Why not just generating an array of random bytes and encoding it to base64 or base64url to get a string key?

File: samples/CookieSessionSample/RedisCacheTicketStore.cs (hunk 2 above)
Comment (xyting) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: thanks

File: samples/CookieSessionSample/RedisCacheTicketStore.cs (hunk 3 above)
Comment (xyting) [inline samples/CookieSessionSample/RedisCacheTicketStore.cs]: OK, this my first PR, I will modify it and PR again

**Comment by Tratcher** [comment]
You can push to the same branch and it will update the PR.

**Pull request closed without merging**